---
title: "HerbPt1 16S processing"
output: html_notebook
---

Pre-analysis filtering of non-bacterial reads
```{r}
#21300 taxa in original
HerbPt1_PS_Orig
#remove all non-bacterial reads
#21028 taxa in bacteria only ps object. 272 non-bacterial taxa 
ps_bac<-subset_taxa(HerbPt1_PS_Orig, Kingdom == "k:Bacteria")
#create seperate dataframe of archea
#101 archeal taxa
ps_archea<-subset_taxa(HerbPt1_PS_Orig, Kingdom=="k:Archaea")

sample_names(ps_bac)
```

Get number of reads per sample
```{r}
sort(sample_sums(ps_bac))
sort(sample_sums(ps_archea))

rare_bac<-rarefy_even_depth(ps_bac, sample.size = 1000)
sample_sums(rare_bac)

#estimate richenss metrics across samples
richness_estimates_bac<-estimate_richness(rare_bac)

#make plot of original sequencing depth by rarefied observed
plot(sample_sums(ps_bac) ~ richness_estimates_bac$Observed)
plot(sample_sums(ps_bac) ~ richness_estimates_bac$Shannon)
plot(sample_sums(ps_bac) ~ richness_estimates_bac$Chao1)


```

